internal/docker/builder.go

```diff
@@ -328,17 +337,19 @@ func (d *Builder) constructHomeserver(blueprintName string, runner *instruction.
 }
 
 // deployBaseImage runs the base image and returns the baseURL, containerID or an error.
-func (d *Builder) deployBaseImage(blueprintName, hsName, contextStr, networkID string) (*HomeserverDeployment, error) {
+func (d *Builder) deployBaseImage(blueprintName string, hs b.Homeserver, contextStr, networkID string) (*HomeserverDeployment, error) {
+	asIDToRegistrationMap := asIDToRegistrationFromLabels(labelsForApplicationServices(hs))
+
 	return deployImage(
-		d.Docker, d.BaseImage, d.CSAPIPort, fmt.Sprintf("complement_%s", contextStr), blueprintName, hsName, contextStr,
+		d.Docker, d.BaseImage, d.CSAPIPort, fmt.Sprintf("complement_%s", contextStr), blueprintName, hs.Name, asIDToRegistrationMap, contextStr,
 		networkID, d.config.VersionCheckIterations,
 	)
 }
 
 // getCaVolume returns the correct mounts and volumes for providing a CA to homeserver containers.
-func getCaVolume(docker *client.Client, ctx context.Context) (map[string]struct{}, []mount.Mount, error) {
-	var caVolume map[string]struct{}
-	var caMount []mount.Mount
+func getCaVolume(docker *client.Client, ctx context.Context) (string, mount.Mount, error) {
```

Comment:
Please document the return types.

Reply:
Refactored the previous `map[string]struct{}` and string `Volumes` stuff out of the code. See https://github.com/matrix-org/complement/pull/74#discussion_r580642237